### ISSUE TITLE
Release candidate tag mismatches with the release API manifest

### ISSUE THREAD

**Issue by hcindyl** [iree-org/iree#10182]
### What happened?

Sometimes the release candidate tag is different from the return result from the GH release API

For example in https://github.com/iree-org/iree/releases/tag/candidate-20220823.244

The tagged sha is [dc06d95](https://github.com/iree-org/iree/commit/dc06d95ac82aa92b4e9551421bc3fe63183d1c43), however, the target_commitish from the first entry of https://api.github.com/repos/google/iree/releases gives `7e8c8310b6fef512a3a59aa585f2c76fd7b7a275`, and such mismatch also shows on the tag page
![8HPcEKv7JPWAm7o](https://user-images.githubusercontent.com/19485875/186301008-4a93923f-0ba8-426c-ae48-4ac0c508a3cd.png)

### Steps to reproduce your issue

Check https://github.com/iree-org/iree/releases/tag/candidate-20220823.244 to see the mismatch

### What component(s) does this issue relate to?

Other

**Comment by alliemartens** [comment]
@GMNGeoffrey Can you help to triage?

**Comment by ScottTodd** [comment]
Possibly fixed by https://github.com/iree-org/iree/pull/10183 (maybe @hcindyl can confirm after a few releases)

**Comment by hcindyl** [comment]
yeah, that's why I haven't closed this issue yet. Keep it open for a few releases and see how it goes. :)

**Comment by hcindyl** [comment]
Recent 5 published tags all have the correct commitish. Mark the issue as closed.
